### PR TITLE
CSV metrics saving, wandb setup, regularization options for end2end training

### PR DESCRIPTION
**Pull request by mmbajo**
## What
* CSV metrics saving
* Regularization options (dropout, weight decay, attention dropout)
* Wandb setup

I will send the .env file in the discord. I also sent out wandb invites in your respective gatech emails :) 

## TODO
* ReazonSpeech Loader
* LoRA implementation
* Start experiments for e2e

## wandb for collaboration
everyone should see each others runs. please also fillout an appropriate run name when executing an experiment. XD
image

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -1,16 +1,17 @@
 poetry run python3 main.py \
     --model_name_or_path="openai/whisper-tiny" \
     --dataset_config_path="conf/dataset_config.yaml" \
+    --wandb_run_name="test_run" \
```
Comment (mmbajo) [inline run.sh:4]: :memo: experiment run name defined here

File: run.sh [hunk 2]
```diff
@@ -22,9 +23,13 @@ poetry run python3 main.py \
...
 	--text_column_name="sentence" \
 	--freeze_feature_encoder="False" \
-	--report_to="tensorboard" \
+	--report_to="wandb" \
 	--metric_for_best_model="wer" \
 	--greater_is_better="False" \
+	--weight_decay="0.01" \
+	--dropout="0.1" \
+	--attention_dropout="0.1" \
+	--activation_dropout="0.1" \
```
Comment (mmbajo) [inline run.sh:32]: :memo: regularization options defined here
